### PR TITLE
Extended documentation

### PR DESCRIPTION
**Pull request by ioquatix**
Fixes <https://github.com/socketry/async/issues/73>.

## Contribution

- [ ] I added tests for my changes.
- [ ] I tested my changes locally.
- [x] I agree to the [Developer's Certificate of Origin 1.1](https://developercertificate.org/).

### CODE REVIEW

File: guides/asynchronous-tasks/readme.md [hunk 1]
```diff
@@ -354,11 +364,15 @@ end
...
+Generally, the reactor's event loop will not exit until all tasks complete. This is informed by {ruby Async::Task#finished?} which checks if the current node has completed execution, which also includes all children. However, there is one exception to this rule: tasks flagged as being `transient` ({ruby Async::Node#transient?}).
 
 ### Transient Tasks
 
-Tasks which are flagged as `transient` do not behave like normal tasks.
+Tasks which are flagged as `transient` are identical to normal tasks, except for one key difference: they do not keep the reactor alive. They are useful for operations which are not directly related to applicatino concurrency, but are instead an implementation detail of the application. For example, a task which is monitoring and maintaining a connection pool, pruning unused connections or possibly ensuring those connections are periodically checked for activity (ping/pong, etc). If all *other* tasks are completed, and only transient tasks remain at the root ofthe reactor, the reactor should exit.
```
Comment (olleolleolle) [inline guides/asynchronous-tasks/readme.md:371]:
```suggestion
Tasks which are flagged as `transient` are identical to normal tasks, except for one key difference: they do not keep the reactor alive. They are useful for operations which are not directly related to application concurrency, but are instead an implementation detail of the application. For example, a task which is monitoring and maintaining a connection pool, pruning unused connections or possibly ensuring those connections are periodically checked for activity (ping/pong, etc). If all *other* tasks are completed, and only transient tasks remain at the root of the reactor, the reactor should exit.
```

File: guides/getting-started/readme.md [hunk 2]
```diff
@@ -15,9 +15,25 @@ $ bundle add async
...
+
+An event loop is part of the implementation of a scheduler which is responsible for waiting for events to occur, and waking up fibers when they are ready to run.
+
+### What is a selector?
+
+A selector is part of the implementation of an event loop which is responsible for interacting with the operating system and waiting for specific events to occur. This is often referred to as "select"ing ready events from a set of file descriptors, but in practice has expanded to encompass a wide range of blocking operations.
+
+### What is a reactor?
+
+A reactor is a specific implementatio of the scheduler interface, which includes an event loop and selector, and is responsible for managing the execution of fibers.
```
Comment (olleolleolle) [inline guides/getting-started/readme.md:35]:
```suggestion
A reactor is a specific implementation of the scheduler interface, which includes an event loop and selector, and is responsible for managing the execution of fibers.
```

**Comment by dniman** [comment]
In guides/asynchronous-tasks/readme.md] 3.times should be changed to 1.upto(3)

**Comment by ioquatix** [comment]
Oh yes thanks.

**Pull request merged**